### PR TITLE
Provisioning: initial unit tests push job

### PR DESCRIPTION
**Pull request by MissingRoberto**
**What is this feature?**

Add more unit test coverate for export job.

**Why do we need this feature?**

We want to improve the quality of the feature.

**Who is this feature for?**

[Add information on what kind of user the feature is for.]

**Which issue(s) does this PR fix?**:

Fixes #

**Special notes for your reviewer:**

Please check that:

- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/contribute/release-notes/#how-to-determine-if-content-belongs-in-whats-new), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/contribute/release-notes/) doc.

### CODE REVIEW

File: pkg/registry/apis/provisioning/jobs/export/worker_test.go [hunk 1]
```diff
@@ -0,0 +1,1323 @@
+package export
```
Comment (MissingRoberto) [inline pkg/registry/apis/provisioning/jobs/export/worker_test.go:1]: I will improve the tests together with the code. Too many situations and client initialization, cloning, etc. 

Some edge cases are not covered yet as it got out of hand.

**Review by ryantxu — APPROVED** [review]
LGTM (but did not test) I don't see any behavior change
